overlord/snapstate/component.go

```diff
@@ -39,12 +38,8 @@ import (
 // snap represented by info must already be installed, and all of the components
 // in names should not be installed prior to calling this function.
 func InstallComponents(ctx context.Context, st *state.State, names []string, info *snap.Info, opts Options) ([]*state.TaskSet, error) {
-	if opts.Flags.Transaction != client.TransactionAllSnaps && opts.Flags.Lane != 0 {
-		return nil, errors.New("cannot specify a lane without setting transaction to \"all-snaps\"")
-	}
-
-	if opts.Flags.Transaction == client.TransactionAllSnaps && opts.Flags.Lane == 0 {
-		opts.Flags.Lane = st.NewLane()
+	if err := opts.setDefaultLane(st); err != nil {
+		return nil, err
 	}
 
 	if err := setDefaultSnapstateOptions(st, &opts); err != nil {
```

Comment:
cleanup: would it make sense to turn this also into a method on opts? would it make sense to merge the two? can be done in a follow up

Reply:
maybe generateLane itself could also become a method on Options?